### DISCUSSION TITLE
I need to set the tooltip text for a filter icon based on the state of the filter modal

### DISCUSSION THREAD

**Discussion by Narek1331** [filamentphp/filament#14054] [Help]
### Package

Table builder

### Package Version

v3.0.0

### How can we help you?

 public static function table(Table $table): Table
    {
        return $table
            ->filtersTriggerAction(function(Action $action){
                $action->tooltip('Open Filter')
                ->button()
                ->label('');
            })}
![Screenshot from 2024-08-26 08-43-21](https://github.com/user-attachments/assets/91077b37-90d9-444c-8e65-95c3a971b2b0)
Hi, can you help me with this? I need to set the tooltip text for a filter icon based on the state of the filter modal. When the modal is closed, the tooltip should display "Open Filter," and when the modal is open, the tooltip should display "Close Filter."

**Comment by leandrocfe** [comment]
```php
->extraAttributes([
    'x-data' => '{ open: false, toggle() { this.open = !this.open } }',
    'x-on:click' => 'toggle()',
    'x-tooltip' => new HtmlString('{content: open ? \'Close Filter\' : \'Open Filter\'}'),
])
```

**Reply by Narek1331** [reply]
Thanks

**Reply by leandrocfe** [reply]
Does it work?

**Reply by Narek1331** [reply]
yes, thank you
